python/triton/language/semantic.py

```diff
@@ -230,7 +230,15 @@ def add(input: tl.tensor | numbers.Number, other: tl.tensor | numbers.Number, sa
         input_scalar_ty = input.type.scalar
         other_scalar_ty = other.type.scalar
     if input_scalar_ty.is_ptr():
-        return tl.tensor(builder.create_addptr(input.handle, other.handle), input.type)
+        other_handle = other.handle
+        if other.dtype.is_int_unsigned() and other.dtype.int_bitwidth < 64:
```

Comment:
Hrm so we're still treating `uint64` as if it were signed because there's no higher type to `zext` to. I guess in practice this is fine because nobody will have enough memory to use the full 64-bit offset anyway. Still feels slightly wrong to me though.

Reply:
for 64bits signed or unsigned would do the same thing since the pointer is 64bits, theyvwould both result in a 64bits add. The problem with lower bits type is that we need to extend which is where the signness affect the arithmetic